### ISSUE TITLE
message-queue: implement telemetry layers / modules

### ISSUE THREAD

**Issue by jan-matejka** [jan-matejka/code-golf#42]
Blocked by #41 

- [ ] Python
- [ ] Go
- [ ] Rust
- [ ] C++
- [ ] Haskell

This is about encapsulating telemetry handling code inside the producers into telemetry module

**Comment by jan-matejka** [comment]
Duplicate of #81
